.github/workflows/release-python.yml

```diff
@@ -1,53 +1,85 @@
 ---
-name: Python Release
+name: Release to PyPI
 
 on:
   release:
     types: [created]
 
 jobs:
-  release:
-    name: PyPI Release
+  build_sdist:
+    name: Build source package
     runs-on: ubuntu-latest
     container: fedora:latest
     steps:
     - uses: actions/checkout@v2
 
-    - name: Install dependencies
+    - name: Install/Upgrade dependencies
       run: |
-        sudo dnf -y install dnf-plugins-core
+        sudo dnf -y install dnf-plugins-core python3-pip
         sudo dnf -y builddep libcomps.spec
-        pip install --upgrade pip
-        pip install pytest twine scikit-build
-
-    - name: Build Python sdist
-      run: python3 setup.py sdist
-
-    - name: Build manylinux Python wheels
-      uses: RalfG/python-wheels-manylinux-build@v0.3.3-manylinux2014_x86_64
-      with:
-        python-versions: 'cp36-cp36m cp37-cp37m cp38-cp38 cp39-cp39'
-        build-requirements: 'scikit-build'
-        system-packages: 'make cmake bzip2-devel expat-devel libxml2-devel xz-devel'
+        python3 -m pip install --upgrade pip pytest scikit-build
 
-    - name: Install and Test Python source package
+    - name: Build and install Python source package
       run: |
-        pip install --user dist/*.tar.gz
-        pushd libcomps/src/python/tests/
-        pytest --verbose --color=yes ./
-        popd
+        python3 setup.py sdist
+        python3 -m pip install dist/*.tar.gz
 
-    - name: Install and Test Python universal binary wheel package
+    - name: Run tests
       run: |
-        pip install --user dist/libcomps-*-cp39-cp39-manylinux2014_x86_64.whl
         pushd libcomps/src/python/tests/
         pytest --verbose --color=yes ./
         popd
 
-    - name: Publish packages to PyPI
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.tar.gz
+
+
+  build_bdist:
+    name: Build binary wheels
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    # setup Python for cibuildwheel
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    # for other architectures, see: https://cibuildwheel.readthedocs.io/en/stable/faq/#emulation
+    - name: Build wheels for CPython
+      uses: pypa/cibuildwheel@v2.1.1
       env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{secrets.PYPI_API_TOKEN}}
-      run: |
-        twine upload dist/*.tar.gz
-        twine upload wheelhouse/*-manylinux*.whl
+        CIBW_ARCHS: auto64  # only 64-bit
+        CIBW_SKIP: "pp*"  # no PyPy builds
+        CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014  # centos 7
+        CIBW_BEFORE_ALL: yum install -y make cmake bzip2-devel expat-devel libxml2-devel xz-devel
```

Comment:
I tried using the same strategy to avoid manually specifying these:

```
CIBW_BEFORE_ALL: |
   yum install -y yum-utils epel-release
   yum-builddep libcomps.spec
```

But for some reason it was freezing up on this step and I couldn't figure out why.

Reply:
Sorry for the very late response but I think the freezing could have been caused by waiting for confirmation of the builddep step.
Shouldn't it be: `yum-builddep -y libcomps.spec`?

Reply:
It may well be, yes.